src/libraries/Microsoft.PowerFx.Core/IR/IRTranslator.cs

```diff
@@ -321,6 +322,13 @@ public override IntermediateNode Visit(FirstNameNode node, IRTranslatorContext c
                 Contracts.AssertValue(node);
                 Contracts.AssertValue(context);
 
+                var nodeType = context.Binding.GetType(node);
+
+                if (nodeType.IsUnknown)
```

Comment:
Should we check and fail earlier (in Translate()) if the binding was done in the "deferred" mode?

Reply:
We can't, because even with deferred mode on user can choose to not pass any deferred args.

Reply:
Disagree, even if the user didn’t reference a deferred arg, binding in deferred mode not being able to generate IR is a useful invariant.